### PR TITLE
refactor: use app picker for test login command

### PR DESCRIPTION
**Pull request by aherzog-auth0**
### Description

Refactors `test login` command to use app picker for selecting application. If no applications exist, it will do and notify the user that we are creating a temporary application to test with and will remove it after. 

Screen Shot 2021-04-02 at 10 10 16 AM

Screen Shot 2021-04-02 at 10 10 44 AM

Screen Shot 2021-04-02 at 10 08 13 AM

### References

https://auth0team.atlassian.net/browse/CLI-120

### CODE REVIEW

File: internal/cli/apps.go [hunk 1]
```diff
@@ -804,7 +807,7 @@ func (c *cli) appPickerOptions() (pickerOptions, error) {
 	}
 
 	if len(opts)+len(priorityOpts) == 0 {
-		return nil, errors.New("There are currently no applications.")
+		return nil, ErrNoApps
```
Comment (Widcket) [inline internal/cli/apps.go:810]: Thanks

File: internal/cli/test.go [hunk 2]
```diff
@@ -130,16 +156,16 @@ auth0 test login -c <id> --connection <connection>`,
 					ansi.Faint("Hint:"), inputs.ClientID)
 
 				if err := cli.setFirstCommandRun(inputs.ClientID, commandKey); err != nil {
+					cleanupTempApplication(isTempClient, cli, inputs.ClientID)
 					return err
 				}
 			}
-
+			cleanupTempApplication(isTempClient, cli, inputs.ClientID)
```
Comment (Widcket) [inline internal/cli/test.go:163]: Shouldn't we be calling the cleanup function once with `defer`?

Comment (cyx) [reply internal/cli/test.go:163]: Yeah, probably similar to the cleanup func of the callbacks PR in #218 we want it to always run.

Comment (aherzog-auth0) [reply internal/cli/test.go:163]: Good call, thanks!

File: internal/cli/apps.go [hunk 3]
```diff
@@ -12,6 +12,9 @@ import (
 	"gopkg.in/auth0.v5/management"
 )
 
+// ErrNoApps signifies no applications exist in a tenant
+var ErrNoApps = errors.New("there are currently no applications")
```
Comment (cyx) [inline internal/cli/apps.go:16]: Only thing I'd say is private by default if we don't think other packages will use this error.

Comment (aherzog-auth0) [reply internal/cli/apps.go:16]: Good call, spaced that everything is in cli package

File: internal/cli/test.go [hunk 4]
```diff
@@ -65,29 +72,45 @@ If --client-id is not provided, the default client "CLI Login Testing" will be u
...
+						return fmt.Errorf("Unable to create an app for testing the login box: %w", err)
+					}
+					inputs.ClientID = client.GetClientID()
+					isTempClient = true
+					cli.renderer.Infof("Default test application successfully created\n")
 				}
-				inputs.ClientID = client.GetClientID()
+			} else {
+				inputs.ClientID = args[0]
 			}
```
Comment (cyx) [inline internal/cli/test.go:110]: Yeah probably want `defer cleanupTempApplication(..)` somewhere here instead

**Review by Widcket — APPROVED** [review]
Thanks!

**Pull request merged**